ingenialink/pdo_network_manager.py

```diff
@@ -132,8 +132,11 @@ def run(self) -> None:
         def stop(self) -> None:
             """Stop the PDO exchange."""
             self._pd_thread_stop_event.set()
-            self.join()
             self._net.stop_pdos()
+            # Only join if we're not trying to join the current thread
```

Comment:
Why flip the stop and the join?
I recall having a race condition problem with this and having to flip it.
You want to make sure that no send_receive_processdata will be sent after calling stop_pdos.